### PR TITLE
Migration of references

### PR DESCRIPTION
**Pull request by marcenacp**
Updates: https://github.com/mlcommons/croissant/issues/115

### CODE REVIEW

File: python/ml_croissant/scripts/migrations/migrate.py [hunk 1]
```diff
@@ -5,11 +5,47 @@
 """
 
 import json
```
Comment (ccl-core) [inline python/ml_croissant/scripts/migrations/migrate.py:34]: I assume we don't need tests for migrations, don't we?

**Review by ccl-core — APPROVED** [review]
Thanks!

File: python/ml_croissant/scripts/migrations/migrate.py (hunk 1 above)
Comment (marcenacp) [reply python/ml_croissant/scripts/migrations/migrate.py:34]: This script is not meant to be re-used ever, so I would say no.

File: datasets/birds/birds_by_openml_converter.json [hunk 2]
```diff
@@ -54,1681 +59,3075 @@
...
+            "distribution": "birds.arff"
+          }
         },
         {
           "@type": "ml:Field",
           "name": "audio_ssd5",
           "dataType": "sc:Number",
-          "source": "#{birds.arff/audio.ssd5}"
+          "source": {
+            "dataExtraction": {
```
Comment (ccl-core) [inline datasets/birds/birds_by_openml_converter.json:112]: Much more elegant and easier to read than the previous version!

File: python/ml_croissant/scripts/migrations/migrate.py (hunk 1 above)
Comment (ccl-core) [reply python/ml_croissant/scripts/migrations/migrate.py:34]: SG!

File: python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py [hunk 3]
```diff
@@ -28,6 +31,32 @@ def parse_reference(issues: Issues, source_data: str) -> tuple[str, ...]:
...
+class FileProperty(enum.Enum):
+    """Lists the intrinsic properties of a file that are accessible from Croissant."""
+
+    content = enum.auto()
+    filename = enum.auto()
+    filepath = enum.auto()
+    fullpath = enum.auto()
+
+
+def is_file_property(file_property: str):
```
Comment (ccl-core) [inline python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py:43]: Might be nice to add a test for this if time allows?

Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py:43]: Thanks! Just did it in another PR: https://github.com/mlcommons/croissant/pull/150/commits/3f1b79824fa42c289e38a44fc192905d87101c01

**Pull request merged**